pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -976,6 +983,183 @@ type FSMirroringSpec struct {
 	// +optional
 	Enabled bool `json:"enabled,omitempty"`
+
+	// SnapshotSchedules is the scheduling of snapshot for mirrored images/pools
+	// +optional
+	SnapshotSchedules []SnapshotScheduleSpec `json:"snapshotSchedules,omitempty"`
+
+	// Retention is the retention policy for a snapshot schedule
+	// One path has exactly one retention policy.
+	// A policy can however contain multiple count-time period pairs in order to specify complex retention policies
+	// +optional
+	SnapshotRetention []SnapshotScheduleRetentionSpec `json:"snapshotRetention,omitempty"`
+}
+
+// SnapshotScheduleRetentionSpec is a retention policy
+type SnapshotScheduleRetentionSpec struct {
+	// Path is the path to snapshot
+	// +optional
+	Path string `json:"path,omitempty"`
+
+	// Duration represents the retention duration for a snapshot
+	// +optional
+	Duration string `json:"duration,omitempty"`
+}
+
+// CephFilesystemStatus represents the status of a Ceph Filesystem
+type CephFilesystemStatus struct {
+	// +optional
+	Phase ConditionType `json:"phase,omitempty"`
+	// +optional
+	SnapshotScheduleStatus *FilesystemSnapshotScheduleStatusSpec `json:"snapshotScheduleStatus,omitempty"`
+	// Use only info and put mirroringStatus in it?
+	// +optional
+	// +nullable
+	Info map[string]string `json:"info,omitempty"`
+
+	// MirroringStatus is the filesystem mirroring status
+	MirroringStatus *FilesystemMirroringInfoSpec `json:"mirroringStatus,omitempty"`
+}
+
+// FilesystemMirroringInfo is the status of the pool mirroring
+type FilesystemMirroringInfoSpec struct {
+	// PoolMirroringStatus is the mirroring status of a filesystem
+	// +optional
+	*FilesystemMirroringInfo `json:",inline"`
+	// LastChecked is the last time time the status was checked
+	// +optional
+	LastChecked string `json:"lastChecked,omitempty"`
+	// LastChanged is the last time time the status last changed
+	// +optional
+	LastChanged string `json:"lastChanged,omitempty"`
+	// Details contains potential status errors
+	// +optional
+	Details string `json:"details,omitempty"`
+}
+
+// FilesystemSnapshotScheduleStatusSpec is the status of the snapshot schedule
+type FilesystemSnapshotScheduleStatusSpec struct {
+	// SnapshotSchedules is the list of snapshots scheduled
+	// +nullable
+	// +optional
+	SnapshotSchedules []FilesystemSnapshotSchedulesSpec `json:"snapshotSchedules,omitempty"`
+	// LastChecked is the last time time the status was checked
+	// +optional
+	LastChecked string `json:"lastChecked,omitempty"`
+	// LastChanged is the last time time the status last changed
+	// +optional
+	LastChanged string `json:"lastChanged,omitempty"`
+	// Details contains potential status errors
+	// +optional
+	Details string `json:"details,omitempty"`
+}
+
+// FilesystemSnapshotSchedulesSpec is the list of snapshot scheduled for images in a pool
+type FilesystemSnapshotSchedulesSpec struct {
+	// Fs is the name of the Ceph Filesystem
+	// +optional
+	Fs string `json:"fs,omitempty"`
```

Comment:
Is this deserialized from ceph, or is this crd schema?
```suggestion
	Filesystem string `json:"filesystem,omitempty"`
```

Reply:
It's coming from Ceph.